docs/content/en/docs/contributing.md

```diff
@@ -89,39 +89,74 @@ at the [pull request review stage](#pull-request-review-flow).
+  - If the PR passes checks, we assign a reviewer. If not, we request additional
+    changes to ensure the PR passes CI checks.
 2. **Review**
-
-- A reviewer will check the PR and potentially request additional changes.
-- If a change is needed, the contributor is requested to make a suggested
-  change. Please make changes with additional commits to your PR, to ensure that
-  the reviewer can easily see the diff.
-- If all looks good, the reviewer will approve the PR.
-- This cycle repeats itself until the PR is approved.
-
+  - A reviewer will check the PR and potentially request additional changes.
+  - If a change is needed, the contributor is requested to make a suggested
+    change. Please make changes with additional commits to your PR, to ensure that
+    the reviewer can easily see the diff.
+  - If all looks good, the reviewer will approve the PR.
+  - This cycle repeats itself until the PR is approved.
 3. **Approved**
-
-- **At this stage, you must squash your commits into a single commit.**
-- Once the PR is approved, a GitHub workflow will
-  [check](https://github.com/google/heir/blob/main/.github/workflows/pr_review.yml)
-  your PR for multiple commits. You may use the `git rebase -i` to squash the
-  commits. Pull requests must comprise of a single git commit before merging.
-
+  - **At this stage, you must squash your commits into a single commit.**
+  - Once the PR is approved, a GitHub workflow will
+    [check](https://github.com/google/heir/blob/main/.github/workflows/pr_review.yml)
+    your PR for multiple commits. You may use the `git rebase -i` to squash the
+    commits. Pull requests must comprise of a single git commit before merging.
 4. **Pull Ready**
-
-- Once the PR is squashed into a single git commit, a maintainer will apply the
-  `pull ready` label.
-- This initiates the internal code migration and presubmits.
-- If needed, we may come to you to make some minor changes in case tests fail at
-  this stage. If so, we will request changes from the GitHub UI and the PR must
-  be approved again.
-- At times, it may not be your change, but it may be additional internal lints
-  and layering checks that are not integrated with `bazel` or open-source
-  tooling. We will go ahead and fix this.
-- Once the internal tests pass, we merge the code internally and externally. You
-  will see `copybara-bot` close the PR and merge your commit directly into main.
+  - Once the PR is squashed into a single git commit, a maintainer will apply the
+    `pull ready` label.
+  - This initiates the internal code migration and presubmits.
+  - After the internal process is finished, the commit will be added to `main`
+    and the PR closed as merged by that commit.
+
+### Internal review details
+
+This diagram summarizes the GitHub/Google code synchronization process.  This
+is largely automated by a Google-owned system called Copybara, the configuration
+for which is Google-internal only. This system treats the Google-internal
```

Comment:
I think we can link to the open-sourced copybara tool, I don't know if the tooling configuration totally lines up, but here it is: https://github.com/google/copybara

Reply:
TIL is was open sourced. Added the link.